src/views/FindOrder.vue

```diff
@@ -272,51 +216,53 @@ export default defineComponent ({
     IonInfiniteScrollContent,
     IonItem,
     IonLabel,
-    IonList,
-    IonListHeader,
+    IonMenu,
     IonNote,
     IonPage,
-    IonSearchbar,
     IonSelect,
+    IonSelectOption,
+    IonSearchbar,
     IonThumbnail,
     IonTitle,
     IonToggle,
     IonToolbar,
+    OrderFilters
   },
   computed: {
     ...mapGetters({
       orders: 'order/getOrders',
       getProduct: 'product/getProduct',
       currentFacilityId: 'user/getCurrentFacility',
       getProductStock: 'stock/getProductStock',
-      isScrollable: 'order/isScrollable'
+      isScrollable: 'order/isScrollable',
+      query: 'order/getOrderQuery'
     })
   },
+  data() {
+    return {
+      shippingMethodOptions: ['any'],
+      orderStatusOptions: ['any'],
+      sort: 'orderDate desc',
+      showOrderItems: true,
+      poIds: {} as any
+    }
+  },
   methods: {
-    async getOrders(vSize?: any, vIndex?: any){
+    async sortOrders(value: string) {
+      this.sort = value
+      await this.store.dispatch('order/updateSortOption', this.sort)
+      this.getOrders();
+    },
+    async getOrders(vSize?: any, vIndex?: any) {
       const viewSize = vSize ? vSize : process.env.VUE_APP_VIEW_SIZE;
       const viewIndex = vIndex ? vIndex : 0;
-      const payload = {
-        "json": {
-          "params": {
-            "rows": viewSize,
-            "start": viewSize * viewIndex,
-            "group": true,
-            "group.field": "orderId",
-            "group.limit": 10000,
-            "group.ngroups": true
-          } as any,
-          "query": "*:*",
-          "filter": "docType: ORDER AND orderTypeId: SALES_ORDER"
-        }
-      }
-      if (this.queryString) {
-        payload.json.params.defType = 'edismax'
-        payload.json.params.qf = 'orderId customerPartyName customerPartyId productId internalName'
-        payload.json.params['q.op'] = 'AND'
-        payload.json.query = `*${this.queryString}*`
+
+      const resp = await this.store.dispatch('order/updateQuery', { viewSize, viewIndex, queryString: this.queryString })
```

Comment:
Have separate method for infinite scroll, pass only in that case and skip passing for other cases. updateQuery action should only handle if viewSize is passed with default values

Reply:
Improved the code on infinite scroll.